tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/mdspan_to_view_mapper_kk.hpp

```diff
@@ -22,19 +30,72 @@ struct LayoutMapper<std::experimental::layout_right> {
 };
 
 template<class MDSpan>
-std::enable_if_t<MDSpan::rank()==1, Kokkos::View<typename MDSpan::element_type*>>
+std::enable_if_t<
+  MDSpan::rank()==1
+  && !is_complex_v<typename MDSpan::element_type>,
+  Kokkos::View<typename MDSpan::element_type*>
+  >
 mdspan_to_view(const MDSpan& a) {
-    return Kokkos::View<typename MDSpan::element_type*>(a.data(),a.extent(0));
+  return Kokkos::View<typename MDSpan::element_type*>(a.data(), a.extent(0));
 }
 
 template<class MDSpan>
-std::enable_if_t<MDSpan::rank()==2, Kokkos::View<typename MDSpan::element_type**,
-                                                 typename LayoutMapper<typename MDSpan::layout_type>::type>>
-mdspan_to_view(const MDSpan& a) {
-    return Kokkos::View<typename MDSpan::element_type**,
-                        typename LayoutMapper<typename MDSpan::layout_type>::type>
-            (a.data(),a.extent(0),a.extent(1));
+std::enable_if_t<
+  MDSpan::rank()==1
+  && is_complex_v<typename MDSpan::element_type>,
```

Comment:
This is incorrect for `const` and/or `volatile`-qualified `element_type`.  Please see my comments on PR #162.